src/Form/fieldUtils.js

```diff
@@ -0,0 +1,53 @@
+import classNames from 'classnames';
+import { useState } from 'react';
+import { newId } from '../utils';
+
+const omitUndefinedProperties = (obj = {}) => Object.entries(obj)
+  .reduce((acc, [key, value]) => {
+    if (value !== undefined) {
+      acc[key] = value;
+    }
+    return acc;
+  }, {});
+
+const callAllHandlers = (...handlers) => {
+  const unifiedEventHandler = (event) => {
+    handlers
+      .filter(handler => typeof handler === 'function')
+      .forEach(handler => handler(event));
+  };
+  return unifiedEventHandler;
+};
+
+const useHasValue = ({ defaultValue, value }) => {
```

Comment:
Tell me about why we need to do this?

Reply:
ultimately this is for floating labels. There is no CSS selector to select inputs or selects that have a value. (and a floating label for an input with a value should be floated above). The one way this can be done is with a `::placeholder-shown` pseudo selector that ie11 does not support. That selector also does not work for select elements since they don't have placeholders. Bootstrap 5 uses this css approach and forces all selects to always float their label above.

TLDR: This provides use the ability to change the behavior of the component based on whether it has a value.

Reply:
_shakes fist at ie11_